code/pilotfile/pilotfile.cpp

```diff
@@ -281,7 +281,7 @@ void pilotfile::update_stats_backout(scoring_struct *stats, bool training)
 
 			for (idx = 0; idx < list_size; idx++) {
 				if ( p_stats->ship_kills[idx].name.compare(Ship_info[i].name) == 0 ) {
-					j = i;
+					j = idx;
```

Comment:
It is feasible that it was a copy-pasta error, the new pilot stats code had a number of bugs and this likely fixes another of them.  Confirmation of that would be helpful though.